core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1253,9 +1253,9 @@ abstract class RDD[T: ClassTag](
 
   /**
    * @return true if and only if the RDD contains no elements at all. Note that an RDD
-   *         may be empty even when it has at least 1 partition.
+   *         may be empty even when it has 1 or more partitions.
    */
-  def isEmpty(): Boolean = partitions.length == 0 || take(1).length == 0
+  def isEmpty(): Boolean = partitions.length == 0 || mapPartitions(it => Iterator(!it.hasNext)).reduce(_&&_)
```

Comment:
I am getting more convinced that this is a bug in `take`. It is provoked by RDD[Nothing].take(1). Take for example these three commands:

~~~ scala
sc.parallelize(Seq[Nothing]()).take(1) // Fails
sc.parallelize(Seq[Any]()).take(1)     // Array[Any] = Array()
sc.parallelize(Seq[Int]()).take(1)     // Array[Int] = Array()
~~~